lib/conduit_async_ssl.mli

```diff
@@ -33,7 +33,7 @@ val ssl_connect :
   ?verify:(Ssl.Connection.t -> bool Deferred.t) ->
   Reader.t ->
   Writer.t ->
-  (Reader.t * Writer.t) Deferred.t
+  (Reader.t * Writer.t) Deferred.Or_error.t
```

Comment:
Well, I'd say they shouldn't under "reasonable" assumptions.
Pipe.create () can fail if there is no fd left for example, but if it's the case you have bigger problems :)